File: CaixaEletronico.js
var readline = require('readline');

var leitor = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

leitor.question("Digite o valor que deseja sacar: ", function (answer) {
    var nota100 = [];
    var nota50 = [];
    var nota20 = [];
    var nota10 = [];
    var saque = parseInt(answer);

    if (saque < 0) {
        console.log("Informe um valor positivo");
    }
    if (isNaN(saque)) {
        console.log("Informe um número");
    }
    if (saque < 10) {
        console.log("Saque apenas acima de 10 reais");
    }
    if ((saque % 10) !== 0) {
        console.log("Valor inválido para saque");
    }
    else {
        do {
            if ((saque - 100) >= 0) {
                saque -= 100;
                nota100.push("nota");
            }
            else if ((saque - 50) >= 0) {
                saque -= 50;
                nota50.push("nota");
            }
            else if ((saque - 20) >= 0) {
                saque -= 20;
                nota20.push("nota");
            }
            else if ((saque - 10) >= 0) {
                saque -= 10;
                nota10.push("nota");
            }
        } while (saque > 0);
    }

    console.log()

    if (nota100.length > 0) {
        console.log("Você receberá " + nota100.length + " nota(s) de R$100,00");
    }
    if (nota50.length > 0) {
        console.log("Você receberá " + nota50.length + " nota(s) de R$50,00");
    }
    if (nota20.length > 0) {
        console.log("Você receberá " + nota20.length + " nota(s) de R$20,00");
    }
    if (nota10.length > 0) {
        console.log("Você receberá " + nota10.length + " nota(s) de R$10,00");
    }
    console.log()

    leitor.close();
});

File: Desafio-DojoPuzzles/CaixaEletronico.py
saque = int(input("Qual valor do saque? "))
print("Serão fornecidas notas de 10, 20, 50 e 100 reais.")

cem = saque//100
cinquenta = saque%100//50
vinte = ((saque % 100) % 50)// 20
dez = (((saque%100) % 50) % 20)//10


if cem != 0:
    print(f"Receberá {cem} notas de R$100,00")
if cinquenta != 0:
    print(f"Receberá {cinquenta} notas de R$50,00")
if vinte != 0:
    print(f"Receberá {vinte} notas de R$20,00")
if dez != 0:
    print(f"Receberá {dez} notas de R$10,00")
print(f"Você sacou R${saque},00")